src/ApiServer/ApiServer/Utilities/DateRangeHelper.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WomPlatform.Web.Api.Utilities;
+
+public class DateRangeHelper {
+    public static string GetDateFormatForRange(DateTime startDate, DateTime endDate)
+    {
+        var totalDays = (endDate - startDate).TotalDays;
+
+        if (totalDays <= 7)
+        {
+            return "%Y-%m-%d"; // Group by day
+        }
+        else if (totalDays <= 365)
+        {
+            return "%Y-%m"; // Group by month
+        }
+        else
+        {
+            return "%Y"; // Group by year
+        }
+    }
+
+    public static IActionResult CheckDateValidity(DateTime startDate, DateTime endDate) {
+        if (endDate < startDate) {
+            return new BadRequestObjectResult("End date cannot be earlier than start date.");
+        }
+        return new OkObjectResult("Date is valid");
+    }
```

Comment:
Ho visto che questo viene usato in giro per il codice (ad esempio in `StatsController::GetTotalVouchersGenerated`) senza considerare il valore di ritorno. Penso sarebbe più utile avere un metodo di verifica che controlla la correttezza della data e solleva un eccezione nel caso in cui ci sia un errore. Ad esempio:

```csharp
public static void CheckDateValidity( ... ) {
    if( /* not valid */ ) {
        throw new ServiceProblemException("End date cannot be earlier than start date", StatusCodes.Status400BadRequest);
    }
}
```

Quel `ServiceProblemException` si può sollevare più o meno dove si vuole in maniera da generare una risposta HTTP con il giusto status code (o almeno così dovrebbe essere).

Se tutto va come deve andare invece il metodo può terminare senza sollevare eccezioni. In questo modo, lato controller, è sufficiente fare:

```csharp
[HttpGet("total-generated-vouchers")]
[Authorize]
[ProducesResponseType(typeof(int), StatusCodes.Status200OK)]
[ProducesResponseType(typeof(void), StatusCodes.Status403Forbidden)]
[ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]
public async Task<ActionResult> GetTotalVouchersGenerated(
    [FromQuery] DateTime startDate,
    [FromQuery] DateTime endDate
) {
    DateRangeHelper.CheckDateValidity(startDate, endDate);

    // ... tutto il resto
```